### PR TITLE
Add MinPointsToBalance to nomination pools

### PR DESCRIPTION
**Pull request by rossbulat**
This PR adds a new runtime constant item for the minimum points to balance ratio to join a nomination pool. Attempts to address #10792.

The name `MinPointsToBalance` has been used as an attempt to keep it short.

fixes https://github.com/paritytech/substrate/pull/11377

The previously hardcoded value of 10, where `OverFlowRisk` has been tested, has been replaced with the storage item.

Tests have been amended to take the new ratio into consideration.

polkadot companion: https://github.com/paritytech/polkadot/pull/5520

### CODE REVIEW

File: frame/nomination-pools/src/lib.rs [hunk 1]
```diff
@@ -1824,6 +1837,7 @@ pub mod pallet {
 			config_op_exp!(MaxPools::<T>, max_pools);
 			config_op_exp!(MaxPoolMembers::<T>, max_members);
 			config_op_exp!(MaxPoolMembersPerPool::<T>, max_members_per_pool);
+			config_op_exp!(MinPointsToBalance::<T>, min_points_to_balance);
```
Comment (ggwpez) [inline frame/nomination-pools/src/lib.rs:1840]: Maybe check for != 0.

**Review by ggwpez — COMMENTED** [review]
Looks good, benchmarks still need to be re-run :+1:

File: frame/nomination-pools/src/lib.rs [hunk 2]
```diff
@@ -706,20 +706,24 @@ impl<T: Config> BondedPool<T> {
...
-		// while restricting the balance to 1/10th of max total issuance,
-		let next_bonded_balance = bonded_balance.saturating_add(new_funds);
-		ensure!(
-			next_bonded_balance <
-				BalanceOf::<T>::max_value().div(MinPointsToBalance::<T>::get().into()),
-			Error::<T>::OverflowRisk
-		);
+		// by `min_points_to_balance`%, if not zero.
+
+		if !min_points_to_balance.is_zero() {
```
Comment (rossbulat) [inline frame/nomination-pools/src/lib.rs:715]: Wondering if this is the best place to check if zero? Can MinPointsToBalance be checked via integrity tests?

File: frame/nomination-pools/src/lib.rs [hunk 3]
```diff
@@ -1136,6 +1144,10 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type SubPoolsStorage<T: Config> = CountedStorageMap<_, Twox64Concat, PoolId, SubPools<T>>;
 
+	/// The minimum points to balance ratio that must be maintained for the pool to be `open`.
```
Comment (kianenigma) [inline frame/nomination-pools/src/lib.rs:1147]: needs an explanation of why `10` would mean a pool with `90%` slash

Comment (rossbulat) [reply frame/nomination-pools/src/lib.rs:1147]: How does the latest commit comment sound?

```
/// The minimum points to balance ratio that must be maintained for the pool to be `open`.
/// This is important in the event slashing takes place and the pool points to balance ratio
/// becomes disproportional.
/// For a value of 10, the threshold would be a pool points to pool balance ratio of 10:1.
/// Such a scenario would also be the equivalent of the pool being 90% slashed.
```

File: frame/nomination-pools/src/lib.rs (hunk 2 above)
Comment (ggwpez) [reply frame/nomination-pools/src/lib.rs:715]: Ah sorry I saw that incorrectly.  
Yea this is not the best spot: It is written in two places: `GenesisConfig::build` and in `set_configs`.  
So it should be checked twice. You should probably also add it to the `sanity_checks`.  

Also in the `is_zero` case you should return an `OverflowRisk` error, right?

Comment (rossbulat) [reply frame/nomination-pools/src/lib.rs]: Post @kianenigma 's advice that I agree with, `MinPointsToBalance` has been changed to a constant. 

We can now check the value is not zero in `integrity_test()`.

**Review by kianenigma — APPROVED** [review]
Thanks!

**Comment by rossbulat** [comment]
> > This PR adds a new storage item for the minimum points to balance ratio
> 
> This PR does not add a new storage item.
> 
> Can you update the description so I can review intention vs code?

yes. The description has been amended.

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
needs companions

**Comment by kianenigma** [comment]
companions need to be happy here.

**Comment by rossbulat** [comment]
> companions need to be happy here.

https://github.com/paritytech/polkadot/pull/5520 Needs another review. Requested @ggwpez

**Comment by ggwpez** [comment]
bot rebase

**Comment by ggwpez** [comment]
bot merge

**Comment by rossbulat** [comment]
bot merge

**Comment by shawntabrizi** [comment]
cumulus CI seems to be breaking here

```
  error[E0046]: not all trait items implemented, missing: `MinPointsToBalance`
      --> /builds/parity/mirrors/substrate/extra_dependencies/polkadot/runtime/kusama/src/lib.rs:1442:1
       |
  1442 | impl pallet_nomination_pools::Config for Runtime {
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `MinPointsToBalance` in implementation
       |
       = help: implement the missing item: `type MinPointsToBalance = Type;`
```

Dunno where `mirrors` is coming from, but it does not seem up to date with the companion

**Pull request merged**